electron/src/js/preload-sso.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+const {webFrame, remote} = require('electron');
+const {SingleSignOn} = remote.require('./js/lib/SingleSignOn');
+
+// Only execute the helper if the origin is the backend
+if (
+  typeof document.location === 'object' &&
+  typeof document!.location!.origin === 'string' &&
+  SingleSignOn.isBackendOrigin(document!.location!.origin)
```

Comment:
Let's check for `if document.location` first, then you probably won't need the `!`.